plugins/meta/iplink/main.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2018 CNI authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is a "meta-plugin". It reads in its own netconf, it does not create
+// any network interface but just changes the network sysctl.
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+
+	"github.com/containernetworking/cni/pkg/skel"
+	"github.com/containernetworking/cni/pkg/types"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/containernetworking/cni/pkg/version"
+	"github.com/containernetworking/plugins/pkg/ns"
+	"github.com/vishvananda/netlink"
+)
+
+/*
+ * Current support ip link command:
+ + ip link set promisc <bool>
+ + ip link set address <mac addr>
+ + ip link set mtu <numeric>
+
+--- syntax
+{
+  "name": "myiplink",
+  "type": "iplink",
+  "iplink": {
+          "promisc": "true",
+          "macaddress": "new mac addr",
+          "mtu": "1454",
+  }
+}
+*/
+
+// TuningConf represents the network tuning configuration.
+type IpLinkConf struct {
+	//SysCtl        map[string]string      `json:"sysctl"`
+	//IpLink        map[string]string      `json:"iplink"`
+	Mac     string `json:"mac,omitempty"`
+	Promisc bool   `json:"promisc,omitempty"`
+	Mtu     int    `json:"mtu,omitempty"`
+}
+
+type PluginConf struct {
+	types.NetConf
+
+	RuntimeConfig struct {
+		IpLinkConf *IpLinkConf `json:"iplink,omitempty"`
+	} `json:"runtimeConfig,omitempty"`
+
+	RawPrevResult map[string]interface{} `json:"prevResult,omitempty"`
+	PrevResult    *current.Result        `json:"-"`
+
+	*IpLinkConf
+}
+
+type MACEnvArgs struct {
+	types.CommonArgs
+	MAC types.UnmarshallableString `json:"mac,omitempty"`
+}
+
+func changeMacAddr(ifName string, newMacAddr string) error {
+	addr, err := net.ParseMAC(newMacAddr)
+	if err != nil {
+		return fmt.Errorf("Invalid args %v for MAC addr: %v", newMacAddr, err)
```

Comment:
Please start message with lower case if it's returned to other func and not displayed directly to user.

Reply:
Fixed